src/frontend/src/components/UserList/index.tsx

```diff
@@ -2,32 +2,57 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { AppState } from '../../redux/store';
 import { loadUsers } from '../../redux/users/actions';
-import { User as UserTypes, UserState } from '../../redux/users/types';
+import { UserState } from '../../redux/users/types';
 import User from '../User';
+import LoadingSpinner from '../LoadingSpinner';
+import NotAuthorized from '../NotAuthorized';
+import TechnicalDifficulties from '../TechnicalDifficulties';
 
 interface Props {
   users: UserState;
-  loadUsers: Function;
+  loadUsers: () => Promise<void>;
+}
+
+interface State {
+  errorType: string;
 }
 
 class UserList extends React.Component<Props> {
-  public componentWillMount() {
+  state: State = {
+    errorType: 'none'
+  };
+  componentDidMount() {
     // this will call the axios request to populate the component with userList
-    this.props.loadUsers();
+    this.props.loadUsers().catch(error => {
+      if (error.response.status === 403) {
+        // error if user is not admin
+        this.setState({ errorType: 'unauthorized' });
+      } else {
+        this.setState({ errorType: 'technical' });
+      }
+    });
   }
 
-  public displayUsers = (inputUsers: UserTypes[]) => {
-    if (inputUsers) {
-      const returnList = inputUsers.map(user => {
-        return <User key={user.id} user={user} />;
-      });
+  displayNoUsers = () => {
+    return this.state.errorType === 'none' ? (
+      <LoadingSpinner inputString={'Users'} />
+    ) : this.state.errorType === 'unauthorized' ? (
+      <NotAuthorized />
+    ) : (
+      <TechnicalDifficulties />
+    );
+  };
+
+  displayUsers = () => {
+    const returnList = this.props.users.userList.map(user => {
+      return <User key={user.id} user={user} />;
+    });
 
-      return returnList;
-    }
+    return returnList;
   };
 
-  public render() {
-    return (
+  render() {
+    return this.props.users.userList.length > 0 ? (
```

Comment:
Nitpick: I think the True case can be wrapped in its own function just like `displayNoUsers`.

Reply:
Simple fix! I will make that change.